static/app/views/alerts/rules/issue/ruleNodeList.tsx

```diff
@@ -254,10 +264,19 @@ class RuleNodeList extends Component<Props> {
 
     const enabledNodes = nodes ? nodes.filter(({enabled}) => enabled) : [];
 
-    const options =
-      selectType === 'grouped'
-        ? groupSelectOptions(enabledNodes)
-        : createSelectOptions(enabledNodes);
+    let options;
+    if (selectType === 'grouped') {
+      options = groupSelectOptions(enabledNodes);
+      if (additionalAction) {
+        for (const option of options) {
+          if (option.label === additionalAction.label) {
+            option.options.push(additionalAction.option);
+          }
+        }
+      }
```

Comment:
Rather than using a `for` loop here, I'd recommend using [`.find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) to get the option, and then modify it.

```js
if (additionalAction) {
  const optionToModify = options.find((option) => option.label === additionalAction.label)
  optionToMoidfy.options.push(additionalAction.option)
} else {
  ...
```
since javascript uses references to variables, this will modify the option in the option list.

---

Using find will also be slightly more efficient as will break out of the loop execution after the option is found.